fury/ui/elements.py

```diff
@@ -3051,3 +3052,118 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
                 self.set_slot_colors()
         i_ren.force_render()
         i_ren.event.abort()
+
+
+class ProgressUI(UI):
+    """Progress Bar UI.
+    """
+
+    def __init__(self, position=(0, 0), size=(300, 100), padding=20,
+                 initial_value=50, min_value=0, max_value=100,
+                 bg_color=(1, 1, 1), progress_color=(0.5, 0.5, 0.5)):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        size : (float, float), optional
+            Center of the slider's center.
+        initial_value : float, optional
+            Initial value of the slider.
+        min_value : float, optional
+            Minimum value of the slider.
+        max_value : float, optional
+            Maximum value of the slider.
+        padding : float, optional
+            Distance between loader and background.
+        bg_color : (float, float, float), optional
+            Background color of progress bar.
+        progress_color : (float, float, float), optional
+            Color of the progress bar.
+        """
+        self.bg_size = size
+        self.padding = padding
+        self.bg_color = bg_color
+        self.progress_color = progress_color
+
+        super(ProgressUI, self).__init__(position)
+
+        self.min_value = min_value
+        self.max_value = max_value
+        self.value = initial_value
+
+        # Offer some standard hooks to the user.
+        self.on_complete = lambda ui: None
+
+        self.update()
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create the ProgressUI with its background (Rectangle2D) and progress
+        (Rectangle2D).
+        """
+        self.progress_total_size = (self.bg_size[0]-2*self.padding,
+                                    self.bg_size[1]-2*self.padding)
+
+        self.background = Rectangle2D(size=self.bg_size, color=self.bg_color)
+        self.progress = Rectangle2D(size=self.progress_total_size,
+                                    color=self.progress_color)
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.background + self.loader
```

Comment:
`self.background.actors` instead of `self.background`. Also, there is no attribute `loader`.